src/gt4py/next/program_processors/runners/dace_iterator/itir_to_sdfg.py

```diff
@@ -189,6 +197,103 @@ def add_storage(
             raise NotImplementedError()
         self.storage_types[name] = type_
 
+    def generate_temporaries(
+        self, node_params: list[Sym], defs_state: dace.SDFGState, program_sdfg: dace.SDFG
+    ) -> dict[str, TaskletExpr]:
+        """Create a table of symbols which are used to define array shape, stride and offset for temporaries."""
+        symbol_map: dict[str, TaskletExpr] = {}
+        # The shape of temporary arrays might be defined based on the shape of other input/output fields.
+        # Therefore, here we collect the symbols used to define data-field parameters that are not temporaries.
+        for sym in node_params:
+            if all([sym.id != tmp.id for tmp in self.tmps]) and sym.kind != "Iterator":
+                name_ = str(sym.id)
+                type_ = self.storage_types[name_]
+                assert isinstance(type_, ts.ScalarType)
+                symbol_map[name_] = SymbolExpr(name_, as_dace_type(type_))
+
+        symbol_dtype = dace.int64
+        tmp_symbols: dict[str, TaskletExpr] = {}
+        for tmp in self.tmps:
```

Comment:
You spend ~70 lines on filling `tmp_symbols` and does not explains what the loop does or the variable represents.
But for the small loop above you write a description, this seems a bit strange.